File: android/app/src/main/java/app/releaf/mobile/features/splash/SplashScreen.kt
/*
 * SplashScreen.kt
 * Full-bleed launch splash derived from the April 2026 Releaf marketing
 * material: deep green, cream-filled diagonal leaf with deep-green
 * vein, lowercase serif wordmark, and the "WRITE. ERASE. REPEAT." loop
 * tagline.
 */

package app.releaf.mobile.features.splash

import androidx.compose.animation.core.LinearEasing
import androidx.compose.animation.core.RepeatMode
import androidx.compose.animation.core.StartOffset
import androidx.compose.animation.core.animateFloat
import androidx.compose.animation.core.infiniteRepeatable
import androidx.compose.animation.core.rememberInfiniteTransition
import androidx.compose.animation.core.tween
import androidx.compose.foundation.Canvas
import androidx.compose.foundation.background
import androidx.compose.foundation.layout.Arrangement
import androidx.compose.foundation.layout.Box
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.Row
import androidx.compose.foundation.layout.fillMaxSize
import androidx.compose.foundation.layout.height
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.layout.size
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.getValue
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.alpha
import androidx.compose.ui.draw.clip
import androidx.compose.ui.geometry.Offset
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.em
import androidx.compose.ui.unit.sp
import app.releaf.mobile.ui.components.ReleafLogoSolid
import app.releaf.mobile.ui.theme.AppColors
import app.releaf.mobile.ui.theme.AppSpacing

@Composable
fun SplashScreen(modifier: Modifier = Modifier) {
    val start = Color(0xFF0F5A35)
    val mid = Color(0xFF0B4328)
    val end = Color(0xFF062F1D)

    Box(
        modifier = modifier
            .fillMaxSize()
            .background(
                Brush.linearGradient(
                    colors = listOf(start, mid, end),
                    start = Offset(0f, 0f),
                    end   = Offset.Infinite,
                ),
            ),
        contentAlignment = Alignment.Center,
    ) {
        SplashDotGrid()

        Column(
            horizontalAlignment = Alignment.CenterHorizontally,
            modifier = Modifier.padding(horizontal = AppSpacing.s6),
        ) {
            ReleafLogoSolid(
                size = 136.dp,
                leafColor = AppColors.OnAccent,
                veinColor = end,
                veinWidth = 4.dp,
            )

            Box(Modifier.height(28.dp))

            Text(
                text = "releaf",
                style = TextStyle(
                    fontFamily = FontFamily.Serif,
                    fontSize = 64.sp,
                    letterSpacing = (-0.015).em,
                ),
                color = AppColors.OnAccent,
            )

            Box(Modifier.height(AppSpacing.s2))

            Text(
                text = "WRITE. ERASE. REPEAT.",
                style = TextStyle(
                    fontFamily = FontFamily.SansSerif,
                    fontSize = 18.sp,
                    letterSpacing = 0.13.em,
                ),
                color = AppColors.OnAccent,
                textAlign = TextAlign.Center,
            )

            Box(Modifier.height(AppSpacing.s3))

            Text(
                text = "Reusable notebook + smart app companion.",
                style = TextStyle(
                    fontFamily = FontFamily.SansSerif,
                    fontSize = 17.sp,
                ),
                color = AppColors.OnAccent.copy(alpha = 0.82f),
                textAlign = TextAlign.Center,
            )

            Box(Modifier.height(72.dp))

            LoadingDots()
        }
    }
}

@Composable
private fun SplashDotGrid() {
    Canvas(Modifier.fillMaxSize()) {
        val step = 18.dp.toPx()
        val radius = 1.dp.toPx()
        var y = 0f
        while (y <= size.height) {
            var x = 0f
            while (x <= size.width) {
                drawCircle(
                    color = Color(0x12F5EEDF),
                    radius = radius,
                    center = Offset(x, y),
                )
                x += step
            }
            y += step
        }
    }
}

@Composable
private fun LoadingDots() {
    val transition = rememberInfiniteTransition(label = "splash-dots")
    Row(horizontalArrangement = Arrangement.spacedBy(AppSpacing.s2)) {
        repeat(3) { index ->
            val alpha by transition.animateFloat(
                initialValue = 0.25f,
                targetValue = 1f,
                animationSpec = infiniteRepeatable(
                    animation = tween(durationMillis = 900, easing = LinearEasing),
                    repeatMode = RepeatMode.Reverse,
                    initialStartOffset = StartOffset(index * 150),
                ),
                label = "splash-dot-$index",
            )
            Box(
                modifier = Modifier
                    .size(8.dp)
                    .alpha(alpha)
                    .clip(CircleShape)
                    .background(AppColors.OnAccent),
            )
        }
    }
}

File: apps/releaf/ios/Releaf/Features/App/SplashScreen.swift
/*
 * SplashScreen.swift
 * Full-bleed launch splash derived from the April 2026 Releaf marketing
 * material:
 *   - Deep green gradient background with faint dot-grid texture.
 *   - Cream-filled diagonal leaf with deep-green vein.
 *   - Lowercase serif "releaf" wordmark.
 *   - "WRITE. ERASE. REPEAT." loop tagline.
 */

import SwiftUI
import ReleafDesignSystem

public struct SplashScreen: View {
    private let brandGreenTop = Color(hex: 0x0F5A35)
    private let brandGreenBottom = Color(hex: 0x062F1D)

    public init() {}

    public var body: some View {
        ZStack {
            LinearGradient(
                colors: [brandGreenTop, Color(hex: 0x0B4328), brandGreenBottom],
                startPoint: .topLeading,
                endPoint: .bottomTrailing
            )
            .ignoresSafeArea()

            SplashDotGrid()
                .ignoresSafeArea()

            VStack(spacing: 0) {
                ReleafLogoSolid(
                    size: 136,
                    leafColor: AppColors.onAccent,
                    veinColor: brandGreenBottom,
                    veinWidth: 4
                )

                Spacer().frame(height: 28)

                Text("releaf")
                    .font(.system(size: 64, weight: .regular, design: .serif))
                    .tracking(-1)
                    .foregroundStyle(AppColors.onAccent)

                Spacer().frame(height: AppSpacing.s2)

                Text("WRITE. ERASE. REPEAT.")
                    .font(.system(size: 18, weight: .bold, design: .default))
                    .tracking(2.2)
                    .foregroundStyle(AppColors.onAccent)
                    .multilineTextAlignment(.center)
                    .padding(.horizontal, AppSpacing.s6)

                Spacer().frame(height: AppSpacing.s3)

                Text("Reusable notebook + smart app companion.")
                    .font(.system(size: 17))
                    .foregroundStyle(AppColors.onAccent.opacity(0.82))
                    .multilineTextAlignment(.center)
                    .padding(.horizontal, AppSpacing.s6)

                Spacer().frame(height: 72)

                LoadingDots()
            }
        }
    }
}

private struct SplashDotGrid: View {
    var body: some View {
        Canvas { context, size in
            let step: CGFloat = 18
            var path = Path()
            var y: CGFloat = 0
            while y <= size.height {
                var x: CGFloat = 0
                while x <= size.width {
                    path.addEllipse(in: CGRect(x: x, y: y, width: 2, height: 2))
                    x += step
                }
                y += step
            }
            context.fill(path, with: .color(AppColors.onAccent.opacity(0.07)))
        }
    }
}

private struct LoadingDots: View {
    @State private var phase: Double = 0

    var body: some View {
        HStack(spacing: AppSpacing.s2) {
            ForEach(0..<3, id: \.self) { index in
                Circle()
                    .fill(AppColors.onAccent)
                    .frame(width: 8, height: 8)
                    .opacity(opacity(for: index))
            }
        }
        .onAppear {
            withAnimation(.linear(duration: 0.9).repeatForever(autoreverses: true)) {
                phase = 1
            }
        }
    }

    private func opacity(for index: Int) -> Double {
        let offset = Double(index) * 0.15
        let raw = (phase + offset).truncatingRemainder(dividingBy: 1)
        return 0.25 + 0.75 * abs(sin(raw * .pi))
    }
}

#Preview {
    SplashScreen()
}
